qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,95 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+
+from .base import BaseEstimatorV2
+from .containers import EstimatorPub, EstimatorPubLike, PrimitiveResult, PubResult
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+class StatevectorEstimator(BaseEstimatorV2):
+    """
+    Simple implementation of :class:`BaseEstimatorV2` with full state vector simulation.
```

Comment:
Please keep in mind that the reno PR will add an example to this docstring. But in addition to that, we could add this, or anything else that you suggest:

```suggestion
    Simple implementation of :class:`BaseEstimatorV2` with full state vector simulation.
    
    This class is implemented via :class:`~.Statevector` which turns provided circuits into 
    pure state vectors. These states are subsequently acted on by :class:~.SparsePauliOp`,
    which implies that, at present, this implementation is only compatible with Pauli-based
    observables.
```

Reply:
Thanks, TBH, I didn't improve the documentation since your reno PR will improve it. Sorry for my lazy work.

Reply:
No, that's okay, I expected the change to happen in the PR, not here. Just trying to get through the comments so we can merge.